### PR TITLE
 🌱 Bump minimum go version to v1.20 and kubectl to v1.28.1

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Review by lentzi90 — APPROVED** [review]
/approve

File: hack/ensure-kubectl.sh
Comment (tuminoid) [inline hack/ensure-kubectl.sh:22]: Q: Should we have this 1.27.x to support skew +/- 1?

**Comment by tuminoid** [comment]
/cherry-pick release-1.5

Let's see if cherry-pick bot is OK with 1.5 branches.

**Comment by lentzi90** [comment]
/hold
I think @tuminoid is correct. We need to have v1.27 for kubectl
Also seems like metal3-dev-env installs go1.19 :thinking:

**Comment by tuminoid** [comment]
Discussed and decided to go with 1.28 anyways.

/override test-centos-integration-release-1-5 
1.5 is not out, so need to skip this to get this merged and unblock the release activity.

/test-centos-e2e-integration-main
Needs to pass though.

**Comment by tuminoid** [comment]
/test-centos-e2e-integration-main 

Prow please wake up.

**Comment by tuminoid** [comment]
Closing and reopening, hoping to revive this PR.

/close

**Comment by tuminoid** [comment]
I think this PR is just dead to Jenkins. Maybe the server reboot caught it in a bad spot, and it keeps thinking the job is already running and won't do anything anymore.

**Comment by kashifest** [comment]
> I think this PR is just dead to Jenkins. Maybe the server reboot caught it in a bad spot, and it keeps thinking the job is already running and won't do anything anymore.

Fooling works  https://github.com/metal3-io/cluster-api-provider-metal3/pull/1165

**Comment by kashifest** [comment]
/close in favor of #1165 
@lentzi90 can you please review the new one

**Pull request closed without merging**
